### DISCUSSION TITLE
NMSCD home page redesign

### DISCUSSION THREAD

**Discussion by Khaoz-Topsy** [NMSCD/community#8] [Ideas]
Hey everyone 👋

The `nmscd.com` domain name needed renewing recently and I thought that it is about time for a redesign of the homepage 😋. Is there anything specific that you would like to see on the home page?

- I want to focus on making the page less "noisy" than the current one, a more flatter and cleaner style. 
- I want to avoid using an HTML template this time, they always seem to be more effort than they are worth 😅. 
- I am also going to have the data of the site separate from the actual HTML, so that it is easier to add and edit projects 😋. 
- I also want to display the current members, I think it would be a nice touch that is missing from the current home page

Let me know if you have other ideas, or if you want to contribute 😋. I will probably create a new repo called `NMSCD.github.io-redesign` so that I can deploy the "new" site to `redesign.nmscd.com` and  leave the existing site and repo alone 😋

_Also I am using this Discussions feature just because I set it up long ago and we haven't used it at all 😅_

**Comment by Lenni009** [comment]
https://nmscd.com/projects.html currently doesn't list all projects. Of course this problem is also caused by the people not editing the page to add their projects (and therefore also shame on me 😅), but I think there's also the formatting hurdle, the fear that it might not look as intended, or that you didn't handle the build step correctly or anything like that. At least that was always my concern when I was less experienced.

I think a JSON file with an array containing objects that list stuff like filepath for image, up to two links and a description would greatly simplify everything. That file could then be looped over and the page generated from it at runtime (or even at build time if that is possible without some major hassles, I don't know).

IIRC last time I edited I also had to run the SCSS command myself? Anyway, not ideal. That should be done by the CI/CD pipeline, not by the user.
I don't know what tech stack you want to use for re-writing the page, but I would strongly suggest using Vite as the build tool. It never disappointed me so far, and I think you also have built some things with it by now.

**Reply by Khaoz-Topsy** [reply]
Yeah! Keeping the data separate from the HTML will make it a lot simple to edit 😁

**Comment by Lenni009** [comment]
Another note on https://nmscd.com: The dev talks might be a little outdated, not sure if you want to keep them on the page. If you want to keep them, maybe move them to a separate page

**Reply by Khaoz-Topsy** [reply]
True, those cringe-y vids don't need to be displayed anywhere😅

**Reply by cyberpunk2350** [reply]
Tbf you could always start doing them again...

**Comment by Khaoz-Topsy** [comment]
Hey @Lenni009 , do you have a sort of icon for your "NMS Archive" and "NMS Coordinate Converter" repositories? I tried making ones for them but they are kind of abstract and hard to represent in a picture/logo 😅. 
![image](https://github.com/NMSCD/community/assets/15005470/696d351a-57e4-4ee8-914c-3cb5928f4b06)

**Reply by Lenni009** [reply]
No, I don't have any icon for them.

I'm not great with graphic design, so I never made anything for these repos 😅

**Reply by Khaoz-Topsy** [reply]
I will make some basic images, maybe using the 📦 and 📐 emojis... 🤔

**Reply by Khaoz-Topsy** [reply]
Now there are tons of icons to make 😂

**Reply by Lenni009** [reply]
Could just use the standard NMSCD icon as unknown.png

**Comment by Khaoz-Topsy** [comment]
The redesign is live [here](https://redesign.nmscd.com). Lenni009 already noticed and made 2 PRs 😂 

I didn't add all the repositories/projects to the website, feel free to suggest or make PRs for repos/projects you want added. The order is also a bit of a tricky thing, I am obviously biased, so please do suggest or make PRs to change up the order (projects are listed in the [projects.json](https://github.com/NMSCD/NMSCD.github.io-Redesign/blob/main/template/data/projects.json) file)

**Reply by Lenni009** [reply]
3 PRs

**Reply by Lenni009** [reply]
We should also implement a "Sqash before merge" or "Rebase to merge" policy or smth like that, otherwise the dev branch could get cluttered with `Merge pull request [...] from [...]`. I never worked with squash or rebase before, so idk what the proper procedure is.

**Reply by Lenni009** [reply]
Or maybe I'm overthinking all this and thinking way too big for a small project website 😅

**Comment by Khaoz-Topsy** [comment]
I have pushed everything to the main repo, archived the redesign repo and cleared the Cloudflare cache of [nmscd.com](https://nmscd.com). I hope you like the pictures I used for your NMSCD projects @Lenni009 😋. 

I made a few SEO improvements as well, resizing of images, updating meta tags, etc. I will check the SEO scores on a few websites again later to see if the improvements made it or not 😅. But the site should be working well now 😁. Let me know if you find something either by creating an [issue on the main repo](https://github.com/NMSCD/NMSCD.github.io/issues) or by creating a PR with the fix in it 😉.

Also mentioning @cyberpunk2350 , just for the funs 🥳

**Reply by Khaoz-Topsy** [reply]
https://suite.seotesteronline.com/seo-checker/aHR0cHM6Ly9ubXNjZC5jb20=/ 🔥🔥🔥

![image](https://github.com/NMSCD/community/assets/15005470/11e5126a-d6d8-4c55-8bff-e516f3b41ada)

**Reply by cyberpunk2350** [reply]
Lol was wondering why my phone pinged
